### PR TITLE
fix: [CI-1600]: Changed error code when gcr hostname is not given

### PR DESCRIPTION
**Pull request by raghavharness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32282)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/ImageSecretBuilder.java [hunk 1]
```diff
@@ -177,7 +177,10 @@ private ImageCredentials getGCRCredentials(String imageName, ConnectorDetails co
     String[] imageParts = imageName.split(PATH_SEPARATOR);
     if (imageParts.length == 0 || !imageParts[0].endsWith(BASE_GCR_HOSTNAME)) {
       throw new InvalidArgumentsException(
-          format("Invalid image: %s for GCR connector", imageName), WingsException.USER);
+              format("Invalid image: %s for GCR connector. Please provide a fully qualified name in the format "
```
Comment (shubham149) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/ImageSecretBuilder.java:180]: Currently, I guess we are not supporting @IMAGE_DIGEST in the image. Can you test the image with digest scenario in a prod pipeline?

Comment (raghavharness) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/ImageSecretBuilder.java:180]: @shubham149 Yeah I tried running with image digest as well and it works.
https://app.harness.io/ng/#/account/9UuUfLwaQ-6ZowvbG7qtLQ/ci/orgs/default/projects/RaghavTest/pipelines/GCR_Test2/executions/T4xlmQbXRw62oYS8PexiLQ/pipeline?stage=_6JT5ga1QJK9ay4tJ80ddw&step=0VxteiiFTBW2p5L3MXZXXw

**Review by shubham149 — APPROVED** [review]

**Comment by shubham149** [comment]
Fix the PR checks that are failing. Then, you can merge the PR.

**Comment by raghavharness** [comment]
trigger messagecheck

**Comment by raghavharness** [comment]
trigger messagecheck

**Comment by raghavharness** [comment]
@shubham149 I have fixed the failing checks can you pls approve again. Also **code-review/reviewable**  check is still in pending state, any idea how to proceed with that ?

**Review by shubham149 — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32282)* status: :shipit: complete! ✓

**Pull request merged**